baybe/targets/numerical.py

```diff
@@ -92,9 +92,9 @@ def _validate_bounds(self, _: Any, value: Interval) -> None:  # noqa: DOC101, DO
         """
         # IMPROVE: We could also include half-way bounds, which however don't work
         # for the desirability approach
-        if not (value.is_finite or not value.is_bounded):
+        if bounds.is_half_open:
             raise ValueError("Bounds must either be finite or infinite on *both* ends.")
```

Comment:
I would also rephrase this to "Bounds are not allowed to be half-open" or similar as the current message is a bit hard to understand.

Reply:
"half-open" was wrong to start with, due to my own confusion with the nomenclature. Please see if the new message makes sense and resolve if OK